app/javascript/components/bulk_edit/CaseStatus.js

```diff
@@ -0,0 +1,237 @@
+import React from 'react';
+import { Button, Modal, Form } from 'react-bootstrap';
+import { PropTypes } from 'prop-types';
+import axios from 'axios';
+import reportError from '../util/ReportError';
+
+class CaseStatus extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showModal: false,
+      patients: [],
+      case_status: '',
+      message: '',
+      confirmed: '',
+      isolation: '',
+      apply_to_group: false,
+      monitoring: false,
+      monitoring_reason: '',
+      public_health_action: '',
+      loading: false,
+    };
+    this.submit = this.submit.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.toggleCaseStatusModal = this.toggleCaseStatusModal.bind(this);
+
+    window.bulkEditCaseStatusComponent = this;
+  }
+
+  activate(patients) {
+    if (!patients || !patients.length) {
+      return;
+    }
+
+    this.getCommonAttributes(patients);
+
+    this.setState({
+      showModal: true,
+      patients: patients,
+    });
+  }
+
+  toggleCaseStatusModal() {
+    let current = this.state.showModal;
+    this.setState({
+      showModal: !current,
+    });
+  }
+
+  // Creates a dictionary, tracking the count of common attributes. This list is used to pre-populate
+  // the form fields and can be displayed to the user for knowledge of the selection.
+  getCommonAttributes(patients) {
+    if (patients.length == 0) return '';
+    var dict = {
+      case_status: {},
+      monitoring: {},
+      isolation: {},
+    };
+    for (var i = 0; i < patients.length; i++) {
```

Comment:
this is more of an overall code style/readability thing, but would probably recommend something more like this:

https://codeburst.io/javascript-array-distinct-5edc93501dc4

(see the 'Distinct property values of an array of objects' and 'Distinct objects by property value from an array of objects' sections)

Reply:
That link worked perfectly, thank you for that! I had at one point wanted to display the count of each type selected, which is why I had used a dict to track the count, but seeing as I never did this is much better!

Reply:
awesome, glad it worked out!